pandas/core/indexes/timedeltas.py

```diff
@@ -406,6 +406,11 @@ def intersection(self, other, sort=False):
         """
         return super().intersection(other, sort=sort)
 
+    def difference(self, other, sort=None):
+        new_idx = super().difference(other, sort=sort)
+        new_idx.freq = None
+        return new_idx
```

Comment:
could this be shared in one of the datetimelike classes?

Reply:
I don't think so: PeriodIndex doesn't allow ``freq=None` and ``DatetimeIndexOpsMixin`` is shared with that`. Don't know if that's a bug in PeriodIndex or as intended, as I haven't worked much with datetime objects.

My motivation with this is getting kwargs out of a basic method and make child classess just add using ``super`` + the specific stuff. That is easier to reason about IMO.